File: Comments.Infrastructure/Models/Comment.cs
using System;
using System.Collections.Generic;

namespace Comments.Infrastructure.Models
{
    public class Comment  
    {
        public int CommentId { get; set; }
        public string CommentText { get; set; }
        public DateTime CreatedDate { get; set; }
        public User User { get; set; }
        public int? HeadCommentId { get; set; }
        public Comment HeadComment { get; set; }        
        public List<Comment> Answers { get; set; }        
        public ICollection<FileStorage> Files { get; set; }                 
    }
}

File: CommentsUI/src/app/comments-page/models/Comment.ts
import { User } from "src/app/shared/models/User";
import { FileStorage } from "./FileStorage";

export class Comment {
    commentId: number;
    headCommentId: number | null;
    user: User;
    createdDate: Date;
    commentText: string;
    answers: Comment[];
    files: FileStorage[];

    constructor(){
        this.commentId = 0;
        this.user = new User();
        this.commentText = "";
        this.answers = new Array<Comment>();
        this.files = new Array<FileStorage>();
    }
}